torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,416 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+def download(url, raw_folder, dataset_name):
+    """Download the dataset if it doesn't exist in processed_folder already."""
+
+    train_csv_path = os.path.join(raw_folder,
+                                  dataset_name + '_csv',
+                                  'train.csv')
+    test_csv_path = os.path.join(raw_folder,
+                                 dataset_name + '_csv',
+                                 'test.csv')
+    if os.path.exists(train_csv_path) and os.path.exists(test_csv_path):
+        return
+
+    os.makedirs(raw_folder)
+    filename = dataset_name + '_csv.tar.gz'
+    url = url
+    path = os.path.join(raw_folder, filename)
+    download_from_url(url, path)
+    extract_archive(path, raw_folder, remove_finished=True)
+
+    print('Dataset %s downloaded.' % dataset_name)
+
+
+def text_normalize(line):
+    """Normalize text string and separate label/text."""
+
+    line = line.lower()
+    label, text = line.split(",", 1)
+    label = "__label__" + re.sub(r'[^0-9\s]', '', label)
+    text = re.sub(r'[^a-zA-Z0-9\s]', ' ', text)
+    text = ' '.join(text.split())
+    line = label + ' , ' + text + ' \n'
+    return line
+
+
+def preprocess(raw_folder, processed_folder, dataset_name):
```

Comment:
```
for filename in os.listdir('dirname):
    with open(filename, 'r') as a, open(filename + '.processed', 'w') as b:
        b.write(text_normalize(a.readline())
```

Reply:
I'd delete this and expect the user to know how to do this

Reply:
Maybe make it an example / add it to the tutorial

Reply:
I would like to support the one-line data loading for the supervised learning dataset. I think we can just leave this function here for these supervised learning datasets.